src/test/ui/rfc-2497-if-let-chains/chains-without-let.rs

```diff
@@ -0,0 +1,18 @@
+fn and_chain() {
+    let z;
+    if true && { z = 3; true} && z == 3 {}
```

Comment:
This is the relevant test

Reply:
For others, [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ef19b5299c8a9c078b063081ee7531ce)'s what this does on stable today:
```
error[E0381]: use of possibly-uninitialized variable: `z`
 --> src/lib.rs:3:34
  |
3 |     if true && { z = 3; true} && z == 3 {}
  |                                  ^ use of possibly-uninitialized `z`
```

I do like that this desugars `&&` to separate `if`s -- it makes sense to me that way -- but I also feel like that should be feature-gated initially.  If not it needs a stabilization report and FCP, I think.

Reply:
Said otherwise, @c410-f3r, I'm wary of merging something that's [self-described](https://github.com/rust-lang/rust/pull/88642#issuecomment-1009861864) as an MVP when reverting it would be a breaking change for ungated code.

However you want to resolve that worry is fine, though -- maybe that's adding a gate, maybe that's restricting this change to things involving `let`s temporarily, maybe that's something else.  As I said, I do *want* this CFG change; it's just a process question.

Reply:
Thanks for the comments, @scottmcm